content/blog/elm-warrior-walkthrough/index.md

```diff
@@ -0,0 +1,818 @@
+---
+title: A walkthrough of Elm Warrior
+date: "2020-08-26T12:00:00.000Z"
+description: Or how to build a simple AI that's slick and thick.
+---
+
+I've recently bumped into [Elm
+Warrior](https://github.com/Skinney/elm-warrior), an Elm adaptation of
+[Warrior JS](https://warriorjs.com/). The premise of the game is to write a
+simple AI that is able to navigate through multiple levels of a dungeon.
+This is how the first level looks like:
+
+![Elm Warrior level one](./level-one.png)
+
+You (the warrior) are represented by the `@`. You always start on the blue
+tile and your goal is to reach the green tile. I thought it would be a cool
+idea to write down how to approach solving a problem like this.
+
+Before we start, a word of warning...
+
+## SPOILERS AHEAD!
+## SPOILERS AHEAD!!
+## SPOILERS AHEAD!!!
+
+Now that my conscience is clear, I'll start by copying the exposed values
+and functions from each top-level module and pasting them all together. No
```

Comment:
I totally skipped the link to Elm Warrior as soon as I came across it, and believe that most readers are very likely to do the same (after all it's in the first sentence of the post and I wanted to continue reading to get an impression of what this was all about).

When I reached this point, I assumed the types you are referring to were the result of your own initial modelling/exploration of the problem or a sneak peek of your final solution instead of the base code provided by the library, and was pretty confused.

I think having a really short description of how the game is set up (ie: you get the modelling of the problem and UI for free and just need to provide a `takeTurn` function) would make everything that comes next much easier to understand.

Reply:
Thank you, I've added a small paragraph that nudges the reader to read https://dev.to/robinheghan/getting-started-with-elm-warrior-5b2n first 👍